### PR TITLE
Kicking people to the side now stuns for a third of the remaining knockdown time.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request
Title

## Why It's Good For The Game
Shove stuns are kind of OP as hell. I've heard headmins asked for it to be long enough to cuff people as a means to stop tiders going into departments, but this _kind of_ impacts other areas of the game? It's just way too long for a no-effort stun that allows you to hit someone with a toolbox 5 times for free before they can stand up (only to be shoved into the wall again).

## Changelog
:cl:
balance: Kicking people to the side now stuns for a third of the remaining knockdown time.
/:cl:

### CODE REVIEW

**Comment by Pugie** [comment]
Flashbang buff

**Comment by HomocidlePancakes** [comment]
Barely changes anything

**Comment by necromanceranne** [comment]
This is a baton buff anything isn't it?

**Comment by Angustmeta** [comment]
Knockdown times from stun batons and most other sources are pretty short, so this heavily nerfs most of them by making floor shove+cuff impossible in most cases. Flashbang is an exception.

**Comment by anconfuzedrock** [comment]
laugh if you want, but I got shoved and kicked to death by an assistant as warden because I gave them a second of benefit of the doubt. probably shouldn't be possible

**Comment by necromanceranne** [comment]
Floor shoves are terrifyingly powerful I am absolutely not arguing anything different. You can currently get people into death spirals as a result of them using tables or wall throws.

**Comment by ATH1909** [comment]
Imagine getting so mad that you got dunked on by an unarmed attacker that you decide to gut shove+kickstunning.

... Except for flashbangs, of course, which now become much stronger with this PR.

**Comment by necromanceranne** [comment]
I mean I've been abusing the shit out of them and have killed people completely unarmed fairly handedly so make of that what you will.

**Comment by carlarctg** [comment]
if flashbangs turn great then honestly good, they're not that useful right now

**Comment by actioninja** [comment]
just remove the stun entirely

**Comment by daklj** [comment]
**keep unrobust people away from balance changes**

**Comment by ExcessiveUseOfCobblestone** [comment]
why does this need a stun?

**Comment by PKPenguin321** [comment]
So you can't disarm them right back while you're prone on the floor maybe? Really hard to drag someone out of your department if you have literally no way of stopping them from shoving you away every time they click short of literally beating them to death

**Comment by ExcessiveUseOfCobblestone** [comment]
you can't have the cake and eat it too though.

You either stop someone nonlethally through the use of hard CCs (which comes with pros/cons of such items existing), or you stop them lethally to the point they're hard CCs (also pros/cons of no such items existing). This of course is assuming they deserved to be stopped, though that gets into admin territory.

I'm a fan of an immediate short hardstun followed by the knockdown, i've even considered entertaining the idea it gets prebuilt into knockdown.

In regards to this though, i'm afraid it might be used to rack up a hardstun which goes against why we changed disarm from a KO to a knockdown in the first place. It should scale with the push knockdowns, be based around a single push, or just be a static number.

basically i'm afraid this PR hits it's goal so well it goes through and hits something else we/author didn't want it to (according to documentation).

**Comment by PKPenguin321** [comment]
>you can't have the cake and eat it too though.

>You either stop someone nonlethally through the use of hard CCs (which comes with pros/cons of such items existing), or you stop them lethally to the point they're hard CCs (also pros/cons of no such items existing), or we need to build some sort of middle ground resolve either via code/admin realm.

Literally, I am not joking, what did you mean by any of this? Doesn't seem to address anything I said and I'm not sure you used that phrase about cake correctly.

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -1348,7 +1348,8 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 				shove_blocked = TRUE
 
 		if(target.IsKnockdown() && !target.IsParalyzed())
-			target.Paralyze(SHOVE_CHAIN_PARALYZE)
+			var/stun_cut = target.AmountKnockdown()/3
```
[the remark this answers is not in the document]
Comment (kriskog) [reply code/modules/mob/living/carbon/human/species.dm]: Old behaviour also had the knockback end at the end of the stun. This happens on line 1356, which I can't add a comment on. I would like you to replace the SHOVE_CHAIN_PARALYZE there with stun_cut.

**Review by kriskog — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
> Literally, I am not joking, what did you mean by any of this? Doesn't seem to address anything I said and I'm not sure you used that phrase about cake correctly.

i have no idea tbh, i think i was having a conversation elsewhere that was a more broad conversation between stuns/lethals and brought it here...

**Comment by necromanceranne** [comment]
@PKPenguin321 the only thing you can do prone is help intent and harm intent. Disarm and Grab don't work.

**Comment by PKPenguin321** [comment]
How is that any better?

**Comment by necromanceranne** [comment]
Okay so why exactly is this closed, because shove stub entirely defeats the purpose of stun removal. This is the strongest stun in the game, and unbelievably powerful by pre-baton stun removal standards even because there have been put in so many knockdowns you can chain knockdowns into shoves to kill someone with ease and you can do this completely unarmed. I've taken down armed opponents with this and abusing microstuns and knockdowns like shove itself.

If the concern is that you can't remove people easily, nonlethally, without a hard stun then the problem is not in this but in other factors. The problem remains that this is a terrible mechanic and will continue to put a hole through any effort for stun removal as it has become the literal crux of all combat.

**Comment by actioninja** [comment]
@optimumtact Wondering about this one as well. Not sure why you closed without any elaboration on to why.

**Comment by optimumtact** [comment]
the headadmins at the time explicitly asked me at the time for a non equipment required stun to help remove people from depts.

I am not going to change it at this time.

**Comment by actioninja** [comment]
Fair enough.
